pynxtools/dataconverter/readers/xps/reader.py

```diff
@@ -225,36 +223,33 @@ def fill_detector_group(key,
                 template[modified_key_unit] = config_dict[f"{key}/@units"]
 
 
-def fill_template_with_xps_data(config_dict,
-                                xps_data_dict,
-                                template,
-                                entry_set):
+def fill_template_with_xps_data(config_dict, xps_data_dict, template, entry_set):
     """Collect the xps data from xps_data_dict
-        and store them into template. We use searching_keys
-        for separating the data from xps_data_dict.
+    and store them into template. We use searching_keys
+    for separating the data from xps_data_dict.
     """
     for key, value in config_dict.items():
-        if XPS_DATA_TOCKEN in value:
-            key_part = value.split(XPS_DATA_TOCKEN)[-1]
-            entries_values = find_entry_and_value(xps_data_dict,
-                                                  key_part,
-                                                  dt_typ=XPS_DATA_TOCKEN)
+        if XPS_DATA_TOKEN in str(value):
+            key_part = value.split(XPS_DATA_TOKEN)[-1]
+            entries_values = find_entry_and_value(
+                xps_data_dict, key_part, dt_typ=XPS_DATA_TOKEN
+            )
 
             fill_data_group(key, entries_values, config_dict, template, entry_set)
 
-        if XPS_DETECTOR_TOCKEN in value:
-            key_part = value.split(XPS_DATA_TOCKEN)[-1]
-            entries_values = find_entry_and_value(xps_data_dict,
-                                                  key_part,
-                                                  dt_typ=XPS_DETECTOR_TOCKEN)
+        if XPS_DETECTOR_TOKEN in str(value):
+            key_part = value.split(XPS_DATA_TOKEN)[-1]
+            entries_values = find_entry_and_value(
+                xps_data_dict, key_part, dt_typ=XPS_DETECTOR_TOKEN
+            )
 
             fill_detector_group(key, entries_values, config_dict, template, entry_set)
 
-        if XPS_TOCKEN in value:
-            tocken = value.split(XPS_TOCKEN)[-1]
-            entries_values = find_entry_and_value(xps_data_dict,
-                                                  tocken,
-                                                  dt_typ=XPS_TOCKEN)
+        if XPS_TOKEN in str(value):
+            token = value.split(XPS_TOKEN)[-1]
+            entries_values = find_entry_and_value(
+                xps_data_dict, token, dt_typ=XPS_TOKEN
+            )
```

Comment:
Can you try to `elif` instead of `if`, we can skip unnecessary check once we get the expected `value`?

Reply:
done